lib/cli/watch-run.js

```diff
@@ -348,15 +360,26 @@ const eraseLine = () => {
 /**
  * Blast all of the watched files out of `require.cache`
  * @param {FSWatcher} watcher - chokidar FSWatcher
+ * @param {boolean} blastAll
  * @ignore
  * @private
  */
-const blastCache = watcher => {
-  const files = getWatchedFiles(watcher);
-  files.forEach(file => {
-    delete require.cache[file];
-  });
-  debug('deleted %d file(s) from the require cache', files.length);
+const blastCache = (watcher, blastAll) => {
+  if (blastAll) {
+    Object.keys(require.cache)
+      // Avoid deleting mocha binary (at minimum, breaks Mocha's watch tests)
+      .filter(file => !file.includes('/mocha/bin/'))
```

Comment:
[Bug] What if someone has a structure like `./src/confusing/mocha/bin/hooks.js`? There's nothing stopping folks from doing that and then reporting it as a bug. This'll need to be a more specific filter.